server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -152,6 +169,174 @@ public DenseVectorFieldMapper build(MapperBuilderContext context) {
         }
     }
 
+    enum ElementType {
+
+        INT8(1) {
+
+            @Override
+            public String toString() {
+                return "byte";
+            }
+
+            @Override
+            KnnVectorField createKnnVectorField(String name, float[] vector, VectorSimilarityFunction function) {
+                byte[] bytes = new byte[vector.length];
+
+                int index = 0;
+                while (index < vector.length) {
+                    bytes[index] = (byte) vector[index++];
+                }
+
+                return new KnnVectorField(name, new BytesRef(bytes), function);
+            }
+
+            @Override
+            IndexFieldData.Builder fielddataBuilder(DenseVectorFieldType denseVectorFieldType, FieldDataContext fieldDataContext) {
+                throw new IllegalArgumentException(
+                    "Fielddata is not supported on field ["
+                        + name()
+                        + "] of type ["
+                        + denseVectorFieldType.typeName()
+                        + "] "
+                        + "with element_type ["
+                        + this
+                        + "]"
+                );
+            }
+
+            @Override
+            void checkVectorBounds(float[] vector) {
+                StringBuilder errorBuilder = null;
+
+                for (int index = 0; index < vector.length; ++index) {
+                    float value = vector[index];
+                    double floor = Math.floor(value);
```

Comment:
I think `Math.rint(value) == value` is the fastest way to validate if a float/double is a whole integer.

Reply:
After looking at the `NumberFieldMapper` type of `byte`, I don't think it makes sense to copy their particular parsing steps as they allow rounding by default which is specifically what we want to avoid. We also never allow String parsing to numeric values as part of dense vector either. 

My main takeaway is when `coerce` is set to `false`, the `NumberFieldMapper` type of `byte` uses `%` to check for non-decimal values. Should we do this same thing for consistency?

Reply:
That makes sense to me, to have the same behavior as the `byte` field type when `coerce` is `false`.

This made me realize, maybe we should switch back to the `byte` naming since it's more consistent with the rest of the mapping API! Sorry about that, I had forgotten we had a built-in `byte` numeric type ...

Reply:
No worries! So for naming it would remain exactly as is it now with `element_type` of `float` or `byte`?

Reply:
I switched this to do the same bounds/decimal checking as the byte field type. It just operates on the parsed float array as opposed to an individual double.

Reply:
Sounds good to switch back to `float` and `byte` and use the parsing strategy you describe.